contracts/buyAndBurn/BuyAndBurnSwapImplementation.sol

```diff
@@ -0,0 +1,364 @@
+        (uint160 currentSqrtPriceX96, , , , , , ) = pool.slot0();
+        uint128 currentLiquidity = pool.liquidity();
+
+        // Determine swap direction based on token addresses (Uniswap convention: token0 < token1)
+        bool zeroForOne = tokenIn < tokenOut;
+
+        uint256 amountSwapIn;
+        uint256 amountSwapOut;
+
+        // STEP 1: Quote full amount with singleBatchImpactThreshold
+        ISwapHelper.Quote memory quote = swapHelper.quoteSlippageExactInputSingle(
+            ISwapHelper.QuoteSlippageExactInputSingleParams({
+                tokenIn: tokenIn,
+                tokenOut: tokenOut,
+                fee: params.fee,
+                amountIn: params.amountIn,
+                sqrtPriceX96: currentSqrtPriceX96,
+                liquidity: currentLiquidity,
+                maximumSlippagePercentage: params.singleBatchImpactThreshold
+            })
+        );
+
+        // STEP 2: Decision point
+        if (quote.amountToPay == params.amountIn) {
+            // YES PATH: Quote accepts full amount - swap everything and return
+
+            // Approve if ERC20
+            if (params.tokenIn != VANA) {
+                IERC20(params.tokenIn).forceApprove(address(swapHelper), params.amountIn);
+            }
+
+            // Swap FULL amount with singleBatchImpactThreshold
+            (uint256 amountSwapInUsed, uint256 amountReceived) = swapHelper.slippageExactInputSingle{
+                    value: params.tokenIn == VANA ? params.amountIn : 0
+                }(
+                ISwapHelper.SlippageSwapParams({
+                    tokenIn: params.tokenIn,
+                    tokenOut: params.tokenOut,
+                    fee: params.fee,
+                    recipient: address(this),
+                    amountIn: params.amountIn,
+                    maximumSlippagePercentage: params.singleBatchImpactThreshold
+                })
+            );
+
+            amountSwapIn = amountSwapInUsed;
+            amountSwapOut = amountReceived;
+
+            // Verify all tokenIn was swapped (YES path invariant)
+            spareIn = params.amountIn - amountSwapIn;
+            require(spareIn == 0, "BuyAndBurnSwap: not all swapped in YES path");
+
+            // All tokenOut goes to burn
+            spareOut = amountSwapOut;
+            liquidityDelta = 0;
+
+            // Transfer tokenOut to burn address
+            // Note: SwapHelper returns native VANA when tokenOut == VANA
+            if (spareOut > 0) {
+                if (params.tokenOut == VANA) {
+                    payable(params.tokenOutRecipient).sendValue(spareOut);
+                } else {
+                    IERC20(params.tokenOut).safeTransfer(params.tokenOutRecipient, spareOut);
+                }
+            }
+
+            // Early return - no LP logic
+            return (liquidityDelta, spareIn, spareOut);
+
+        } else if (quote.amountToPay < params.amountIn) {
```

Comment:
Should we just use `else` here because `quote.amountToPay` is always less than or equal `params.amountIn`?